src/main.zig

```diff
@@ -381,6 +383,57 @@ pub fn Deserializer(comptime ReaderType: type) type {
             };
         }
 
+        /// Deserializes an enum (stored as an integer)
+        pub fn deserializeEnum(self: *Self, comptime T: type) ReadError!T {
+            if (@typeInfo(T) != .Enum) @compileError("Expected enum type, but found type '" ++ @typeName(T) ++ "'");
+
+            const I = comptime meta.Tag(T);
+
+            // Read the value as an integer
+            var deserialized_int = try self.deserializeInt(I);
+
+            // Validate that the integer that was read is a valid member of the enum
+            var valid: bool = false;
+
+            inline for (meta.fields(T)) |enum_field| {
+                if (enum_field.value == deserialized_int) {
+                    valid = true;
+                    break;
+                }
+            }
+
+            if (!valid) {
+                return error.MismatchingFormatType;
+            }
+
+            return @intToEnum(T, deserialized_int);
```

Comment:
This can be simplified as:
```zig
const deserialized_int = try self.deserializeInt(I);
return std.meta.intToEnum(T, deserialized_int) catch return error.MismatchingFormatType;
```

Reply:
Thanks, didn't know about this method. Fixed it. 😃